src/SplitPackages/Program.cs

```diff
@@ -0,0 +1,288 @@
+            }
+
+            bool whatif = args.Length == 4 && args[3].Equals("whatif", StringComparison.OrdinalIgnoreCase);
+
+            try
+            {
+                var arguments = NormalizeArguments(args);
+                var packagesFromSource = GetPackagesFromSourceFolder(arguments.SourceFolder);
+                var packagesFromCsv = GetPackagesFromCsvFile(arguments.CsvFile);
+                var expandedPackages = ExpandPackages(packagesFromCsv, arguments.SourceFolder);
+
+                EnsureNoExtraSourcePackagesFound(packagesFromSource, expandedPackages);
+
+                EmitWarningsForMissingPackages(packagesFromSource, packagesFromCsv);
+
+                EnsureDestinationDirectories(
+                    arguments.DestinationFolder,
+                    expandedPackages.Select(e => e.DestinationFolderName).Distinct());
+
+                SetDestinationPath(arguments, expandedPackages);
+
+                CopyPackages(expandedPackages, whatif);
+            }
+            catch (Exception e)
+            {
+                WriteError(e.Message);
+                return;
+            }
+        }
+
+        private static Arguments NormalizeArguments(string[] args)
+        {
+            var sourcePackages = Path.GetFullPath(args[0]);
+            var csvFile = Path.GetFullPath(args[1]);
+            var destinationFolder = Path.GetFullPath(args[2]);
+
+            if (!Directory.Exists(sourcePackages))
+            {
+                var msg = $@"Source packages folder does not exist or is not a folder
+{sourcePackages}";
+                throw new InvalidOperationException(msg);
+            }
+
+            if (!File.Exists(csvFile))
+            {
+                var msg = $@"csv file does not exist or is not a file
+{csvFile}";
+                throw new InvalidOperationException(msg);
+            }
+
+            if (!Directory.Exists(destinationFolder))
+            {
+                var msg = $@"Destination folder does not exist, creating folder at:
+{destinationFolder}";
+                WriteInformational(msg);
+                Directory.CreateDirectory(destinationFolder);
+            }
+
+            return new Arguments
+            {
+                DestinationFolder = destinationFolder,
+                CsvFile = csvFile,
+                SourceFolder = sourcePackages
+            };
+        }
+
+        private static IList<PackageItem> GetPackagesFromSourceFolder(string sourceFolder)
+        {
+            var packages = Directory.EnumerateFiles(sourceFolder);
+
+            return packages.Select(p => new PackageItem
+            {
+                OriginPath = p,
+                Name = Path.GetFileName(p),
+                FoundInFolder = true
+            }).ToList();
+        }
+
+        private static IList<PackageItem> GetPackagesFromCsvFile(string csvFile)
+        {
+            var lines = File.ReadAllLines(csvFile);
+            var parsedLines = lines.Select(l => l.Split(','));
```

Comment:
Yeah this clearly isn't a proper CSV parser, but in our case the inputs are highly constrained.